lib/livereload.js

```diff
@@ -23,111 +24,72 @@ module.exports = LiveReload;
-
-    // Set default options if necessary
-    this.opts = opts;
-    this.opts.openBrowser = opts.openBrowser || false;
-
+function LiveReload(projectRoot, platformWWWs, options) {
     this.projectRoot = projectRoot;
     this.startPage = helpers.GetStartPage(this.projectRoot);
     this.www_dir = path.join(this.projectRoot, 'www');
     platformWWWDirs = platformWWWs;
     this.StartServer = StartServer;
     this.StopServer = StopServer;
+    this.options = options;
+    this.browserSync = require('browser-sync').create(); // If this is exposed to users, how would they handle deps install ?
+    return this;
 };
 
 
 function StartServer() {
-
     var self = this;
     var deferred = Q.defer();
 
-    return Q().then(function() {
-        browserSync = require('browser-sync').create();
-
-        browserSync.watch(path.join(self.www_dir, '**/*.*'), {}, function(event, files) {
-
-            if (event !== 'change') {
-                return;
-            }
-
-            // After a file changes, first run prepare, then reload.
-            return self.prepareFn().then(function() {
-
-                // Remove CSP directive from HTML
-                var CSPRemover = require('./utils/CSPRemover');
-                platformWWWDirs.forEach(function(platDir) {
-                    var platformIndexLocal = path.join(platDir, self.startPage);
-                    var remover = new CSPRemover(platformIndexLocal);
-                    remover.Remove();
-                });
-
-                return Q();
-            }).then(function() {
-                browserSync.reload(files);
-            });
-        });
-
-        // Initialize the browser-sync instance
-        browserSync.init({
-            server: {
-                baseDir: self.projectRoot,
-                directory: true
-            },
-            files: path.join(self.www_dir, '**/*.*'),
-            open: (self.opts.openBrowser || false),
-            snippetOptions: {
-                rule: {
-                    match: /<\/body>/i,
-                    fn: function(snippet, match) {
-                        return monkeyPatch() + snippet + match;
-                    }
+    var defaultOptions = {
+        server: {
+            baseDir: self.projectRoot,
+            directory: true
+        },
+        open: false,
+        snippetOptions: {
+            rule: {
+                match: /<\/body>/i,
+                fn: function(snippet, match) {
+                    return monkeyPatch() + snippet + match;
                 }
             }
-        }, function(err, bs) {
-
-            // Once BrowserSync is ready, resolve the promise
+        }
+    };
 
-            if (err) {
-                deferred.reject(err);
-                return;
-            }
 
-            var server_url = bs.options.getIn(['urls', 'external']);
+    // User provided options take precedence over the default ones
+    self.options = _.merge(defaultOptions, self.options);
```

Comment:
While merging sounds like the simple thing to do - a user of the API if they override some behaviors - it will actually result in issues for example if they remove the monkeypath.